src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -205,8 +208,7 @@ internal override void ClearAllWritePackets()
         {
             if (_sniPacket != null)
             {
-                _sniPacket.Dispose();
-                _sniPacket = null;
```

Comment:
Why remove this line? The Garbage collector will see a null reference and perform a cleanup. Also I would recommend using the Dispose() call because it seems more natural to Dispose and set to null. (even though Dispose and Release do the same thing)

Reply:
With those lines, `_sniPacket` recreated over and over again, which is pointless. It can reuse existing `_sniPacket`

Reply:
Sorry I don't follow. The design looks like `_sniPacket` is taken from the cache and not allocated over and over.

Reply:
And `_sniPacket` object is disposed when `TdsParserStateObject` is disposed.